src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -19,11 +19,11 @@ public class AuthenticationServiceImpl implements AuthenticationService {
     public User login(String email, String password) throws AuthenticationException {
         Optional<User> userFromDbOptional = userService.findByEmail(email);
         User user = userFromDbOptional.get();
-        if (userFromDbOptional.isEmpty() || !(user.getEmail().equals(email))
-                || !(user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt())))) {
-            throw new AuthenticationException("Can't authenticate user");
+        if (userFromDbOptional.isPresent() && user.getEmail().equals(email)
+                && checkPassword(user, password)) {
```

Comment:
```suggestion
        if (userFromDbOptional.isPresent() && checkPassword(user, password)) {
```
check for email is redundant because we use the method `findByEmail` before